### PR TITLE
backport: merge bitcoin#23156, #23213, #23227, #23223, #23564, #23538, #23437, #23630, #23465, #23738, #17631, #22875 (auxiliary backports: part 18)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/6288

* Dependent on https://github.com/dashpay/dash/pull/6296

## Breaking changes

- `quorum dkgsimerror` will no longer accept a decimal value between 0 and 1 for the `rate` argument, it will now expect an integer between 0 to 100.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — DISMISSED** [review]
LGTM 328b14ad8df9155b2a9f37a8fe17ad0981686072 but I'd prefer 0eac8b58965f8def5fd2663f56c8b3b1c36ce670 be non-partial

**Review by PastaPastaPasta — APPROVED** [review]
utACK 5aceee38fcb7ea1221a7dac59578afe77c510735

File: src/rest.cpp [hunk 1]
```diff
@@ -30,6 +32,7 @@
 #include <univalue.h>
 
 static const size_t MAX_GETUTXOS_OUTPOINTS = 15; //allow a max of 15 outpoints to be queried at once
+static constexpr unsigned int MAX_REST_HEADERS_RESULTS = 2000;
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/rest.cpp:35]: meh, prolly fine as is

File: src/rest.cpp [hunk 2]
```diff
@@ -336,6 +339,208 @@ static bool rest_block_notxdetails(const CoreContext& context, HTTPRequest* req,
...
+            }
+
+            return RESTERR(req, HTTP_NOT_FOUND, errmsg);
+        }
+        filter_headers.push_back(filter_header);
+    }
+
+    switch (rf) {
+    case RetFormat::BINARY: {
+        CDataStream ssHeader(SER_NETWORK, PROTOCOL_VERSION);
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/rest.cpp:414]: `RPCSerializationFlags` was introduced in https://github.com/bitcoin/bitcoin/pull/9194 to help with segwit deployment, we don't have it

**Review by UdjinM6 — APPROVED** [review]
utACK 5aceee38fcb7ea1221a7dac59578afe77c510735

**Review by knst — APPROVED** [review]
utACK 5aceee38fcb7ea1221a7dac59578afe77c510735

**Pull request merged**